internal/k8s/controller.go

```diff
@@ -2649,6 +2840,46 @@ func (lbc *LoadBalancerController) addWAFPolicyRefs(
 	return nil
 }
 
+// addBadosPolicyRefs ensures the app protect dos resources that are referenced in policies exist.
+func (lbc *LoadBalancerController) addBadosPolicyRefs(
```

Comment:
Could we have a unit test similar to TestAddWAFPolicyRefs?

Reply:
Add unit test  for AddBadosPolicyRefs